index.bs

```diff
@@ -1771,103 +1678,77 @@ interface MLGraph {
-                                            1. Set the content of |results|[|outputName|].{{MLOutput/data}} to the content of output tensor returned by |this|.{{MLGraph/[[implementation]]}}.
-                                    1. Remove |outputName| from |remainingOutputNames|.
-                                    1. If |remainingOutputNames| is empty, then resolve |promise| with |results| and stop.
-                            </div>
-                </div>
-
-            1. Return |promise|.
+            1. For each |key| -> |value| of |inputs|:
+                1. Let |inputDesc| be |this|.{{MLGraph/[[inputDescriptors]]}}[|key|].
+                1. Let |inputTensor| be a new tensor for |this|.{{MLGraph/[[implementation]]}} of data type that is compatible with |inputDesc|.{{MLOperandDescriptor/type}}.
+                1. If |value| is an {{MLInput}} and |value|.{{MLInput/dimensions}} is given, then:
+                    1. Set the dimensions of |inputTensor| to |value|.{{MLInput/dimensions}}.
+                1. Else:
+                    1. Set the dimensions of |inputTensor| to |inputDesc|.{{MLOperandDescriptor/dimensions}}.
+                1. If |value| is an {{MLInput}}, then:
+                    1. Set the values of |inputTensor| to the values of |value|.{{MLInput/resource}}.
+                1. If |value| is an {{MLResource}}, then:
+                    1. Set the values of |inputTensor| to the values of |value|.
+                1. Set the input of |this|.{{MLGraph/[[implementation]]}} that is associated with |key| to |inputTensor|.
+            1. For each |key| -> |value| of |outputs|:
+                1. Issue a compute request for output of |this|.{{MLGraph/[[implementation]]}} that is associated with |key|.
+                1. Wait for the compute request to be completed.
+                1. If there is an error returned by |this|.{{MLGraph/[[implementation]]}}, then:
+                    1. Throw an {{OperationError}} {{DOMException}} and stop.
+                1. Else:
+                    1. Let |outputTensor| be the output tensor returned by |this|.{{MLGraph/[[implementation]]}}.
+                    1. If the kind of |value| is not compatible with the value type of |outputTensor|, then throw a {{DataError}} {{DOMException}} and stop.
+                    1. Let |outputSize| be 1.
+                    1. For each |dimension| of dimensions of |outputTensor|:
+                        1. Set |outputSize| to the product of |outputSize| and |dimension|.
+                    1. If |outputSize| is greater than the length of |value|, then:
+                        1. Throw a {{DataError}} {{DOMException}} and stop.
+                    1. Else:
+                        1. Set the values of |value| to the values of |outputTensor|.
+            1. Return {{undefined}}.
```

Comment:
It is not common for Web APIs to return undefined. It is more common to return null. I propose we align with that convention.

Reply:
I follow [WebIDL](https://heycam.github.io/webidl/#idl-undefined) that uses `undefined` as `void` to define a function returning nothing. There are two spec examples of this usage: [forEach](https://heycam.github.io/webidl/#es-forEach) and [ApplyConstraints](https://www.w3.org/TR/mediacapture-streams/#dfn-applyconstraints-algorithm)

Reply:
Thanks! That is correct usage. I misunderstood the semantics.

For reference:
https://github.com/heycam/webidl/pull/906
https://github.com/heycam/webidl/issues/60